types/single-spa-react.d.ts

```diff
@@ -1,21 +1,31 @@
 import * as React from "react";
-import * as ReactDOM from "react-dom";
 import { AppProps, CustomProps, LifeCycleFn } from "single-spa";
 
 export const SingleSpaContext: React.Context<CustomProps & AppProps>;
 
+type DeprecatedRenderTypes =
+  | "createBlockingRoot"
+  | "unstable_createRoot"
+  | "unstable_createBlockingRoot";
+
+type LegacyRenderType = "hydrate" | "render";
+
+type RenderType =
+  // React 18
+  "createRoot" | "hydrateRoot" | LegacyRenderType;
+
 export interface SingleSpaReactOpts<RootComponentProps> {
-  React: typeof React;
-  ReactDOM: typeof ReactDOM;
+  React: any;
+  ReactDOM?: {
+    [T in LegacyRenderType]?: any;
+  };
+  ReactDOMClient?: {
+    [T in RenderType]?: any;
+  };
   rootComponent?:
     | React.ComponentClass<RootComponentProps, any>
     | React.FunctionComponent<RootComponentProps>;
-  loadRootComponent?: (
-    props: RootComponentProps
-  ) => Promise<
-    | React.ComponentClass<RootComponentProps, any>
-    | React.FunctionComponent<RootComponentProps>
-  >;
+  loadRootComponent?: (props?: any) => Promise<React.ElementType<any>>;
```

Comment:
This is another `any` that I think could become a real type. Why not continue to use `RootComponentProps` instead of `any`? If not, then I'm not sure what the point is for making the interface a generic.

Please note, I might be missing something here. But at first glance, it doesn't look like the switch to `any` is buying us anything. It's only potentially hiding errors.

Reply:
I agree on this one. I'll update it to use a generic.